### PR TITLE
32 user can back a page

### PR DESCRIPTION
**Pull request by lsaville**
@bermannoah @bfpepper
this:
-is a giant test (could use some splitting or smaller supporting tests
-allows a user to back a project
-adds a method to make the progress bar dynamic
-adds super cool relationships so that you can call @project.backers and it gives you a collection of backing users and @user.backed_projects for a collection of projects the user is backing.
-adds a pledges table that is a join between users and projects (and forms the basis of the aforementioned new relationships)
Check it out and see if it is to your liking 🌮 🎉 💯

### CODE REVIEW

**Comment by antciccone** [comment]
@lsaville what a test that is 👍 good work!!

File: app/controllers/projects/pledges_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Projects::PledgesController < ApplicationController
```
Comment (bermannoah) [inline app/controllers/projects/pledges_controller.rb:1]: Nice namespacing!

File: app/models/project.rb [hunk 2]
```diff
@@ -6,11 +6,19 @@ class Project < ApplicationRecord
...
   has_many :users, through: :user_projects
 
   validates :title,
              :slug, uniqueness: true
 
   belongs_to :category
+
+
+  def funding_received
+    pledges.empty? ? 0 : ((pledges.sum(:amount_given) / goal.to_f) * 100).round
```
Comment (bermannoah) [inline app/models/project.rb:22]: A magical ternary!

File: spec/features/backers/a_user_can_back_a_project_spec.rb [hunk 3]
```diff
@@ -1,21 +1,32 @@
...
-#       #need some more expectations
-#     end
-#   end
-# end
+require 'rails_helper'
+
+describe "As a user" do
+ context "when I visit a projects show page" do
+  scenario "I can back(give money) to a project" do
+      project = create(:project, title: "How to find a Job", goal: 100)
```
Comment (bermannoah) [inline spec/features/backers/a_user_can_back_a_project_spec.rb:6]: it me

**Review by bermannoah — COMMENTED** [review]
Awesome! Good to merge. @bfpepper yer thoughts?

**Pull request merged**